sentry-android-navigation/build.gradle.kts

```diff
@@ -0,0 +1,86 @@
+import io.gitlab.arturbosch.detekt.Detekt
+import io.gitlab.arturbosch.detekt.extensions.DetektExtension
+
+plugins {
+    id("com.android.library")
+    kotlin("android")
+    jacoco
+    id(Config.QualityPlugins.gradleVersions)
+    id(Config.QualityPlugins.detektPlugin)
+}
+
+android {
+    compileSdk = Config.Android.compileSdkVersion
+
+    defaultConfig {
+        targetSdk = Config.Android.targetSdkVersion
+        minSdk = Config.Android.minSdkVersion
+
+        // for AGP 4.1
+        buildConfigField("String", "VERSION_NAME", "\"${project.version}\"")
+    }
+
+    buildTypes {
+        getByName("debug")
+        getByName("release") {
+            consumerProguardFiles("proguard-rules.pro")
+        }
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8.toString()
+    }
+
+    testOptions {
+        animationsDisabled = true
+        unitTests.apply {
+            isReturnDefaultValues = true
+            isIncludeAndroidResources = true
+        }
+    }
+
+    lint {
+        warningsAsErrors = true
+        checkDependencies = true
+
+        // We run a full lint analysis as build part in CI, so skip vital checks for assemble tasks.
+        checkReleaseBuilds = false
+    }
+
+    variantFilter {
+        if (Config.Android.shouldSkipDebugVariant(buildType.name)) {
+            ignore = true
+        }
+    }
+}
+
+tasks.withType<Test> {
+    configure<JacocoTaskExtension> {
+        isIncludeNoLocationClasses = false
+    }
+}
+
+kotlin {
+    explicitApi()
+}
+
+dependencies {
+    api(projects.sentry)
+
+    implementation(Config.Libs.navigationRuntime)
```

Comment:
Quick question, What's about the other 2.x versions? are they all compatible with the public APIs that we use?
I'm wondering if we actually should start using `compileOnly`

Reply:
Yeah looks like it's compatible all the way to version 1.0.0-alpha10 actually. We could use `compileOnly` yes, I think it'd be the right approach, rather than silently updating user's dependency tree to our version (if they have an older one).

Reply:
Maybe start using `compileOnly` for this one and raise an issue for the other dependencies?
We could even have a matrix test in the future that runs with multiple versions of the transitive dependencies, similar to SAGP.